python/lsst/dax/distribution/DataGenConnection.py

```diff
@@ -0,0 +1,727 @@
+
+        Note
+        ----
+        Parameters for servRespInit are the return values for clientRespInit.
+        """
+        sep = self.COMPLEXSEP
+        print("ingest_dict=", ingest_dict)
+        skip_val = '0'
+        if ingest_dict['skip']:
+            skip_val = '1'
+        msg = (name + sep + arg_string + sep + cfg_file_contents
+            + sep + ingest_dict['host'] + sep + str(ingest_dict['port'])
+            + sep + ingest_dict['user'] + sep + ingest_dict['auth']
+            + sep + ingest_dict['db'] + sep + skip_val)
+        self._send_msg(self.S_INIT_R, msg)
+    def clientRespInit(self):
+        """Unwrap the configuration information sent by the server.
+
+        Return
+        ------
+        name : str
+            Name for the client.
+        arg_string : str
+            Argument string for dataGen.py.
+        cfg_file_contents : str
+            Contents of the configuration file for dataGen.py.
+        ingest_dict : dictionary
+            Dictionary with ingest system configuration information
+            where the keys must match those in servRespInit.
+        """
+        msg_id, msg, msg_len = self._recv_msg()
+        if not msg_id == self.S_INIT_R:
+            self.warnings += 1
+            raise DataGenError('ERROR clientRespInit ' + str(msg_id) + ' ' + str(msg_len) + ' ' + msg)
+        sep = self.COMPLEXSEP
+        splt_msg = msg.split(sep)
+        name = splt_msg[0]
+        arg_string = splt_msg[1]
+        cfg_file_contents = splt_msg[2]
+        ingest_dict = { 'host':splt_msg[3], 'port':int(splt_msg[4]), 'user':splt_msg[5], 'auth':splt_msg[6],
+                        'db':splt_msg[7] }
+        skip_val = splt_msg[8]
+        ingest_dict['skip'] = bool(skip_val != '0')
+        return name, arg_string, cfg_file_contents, ingest_dict
+
+    def clientReqPartitionCfgFile(self, index):
+        """Client request a partitioner configuration file from the server
+
+        Parameters
+        ----------
+        index : int
+            Indicates which partitioner configuration file the client wants.
+            The client starts at 0 and keeps incrmenting by 1 until the server
+            indicates the file does not exist. The number of files varies
+            depending on the database.
+        """
+        print("clientReqChunks C_PCFGR")
+        self._send_msg(self.C_PCFG_R, str(index))
+    def servRespPartitionCfgFile(self):
+        """Serv recieve the partition configuration index from the client.
+
+        Return
+        ------
+        index : int
+            Indicates which partitioner configuration file the client wants.
+        """
+        print("servRespPartitionCfgFile C_PCFGR")
+        msg_id, msg, msg_len = self._recv_msg()
+        if not msg_id == self.C_PCFG_R:
+            self.warnings += 1
+            raise DataGenError('ERROR servRespPartitionCfgFile' + str(msg_id) + ' ' + str(msg_len) + ' ' + msg)
+        return int(msg)
```

Comment:
Need blank line.

Reply:
I added lines between the functions for this file.